static/js/starred_messages.js

```diff
@@ -34,6 +34,12 @@ exports.get_starred_msg_ids = function () {
     return Array.from(exports.ids);
 };
 
+exports.get_topic_starred_msg_ids = function (topic_name) {
+    return Array.from(exports.ids).filter(function (id) {
+        return message_store.get(id).subject === topic_name;
```

Comment:
`.topic` should be available here. (Use one of the functions from `util.js` to get `topic` if .topic method is undefined).

Reply:
We are missing check for stream_id here too.
Something like this is more appropriate:
```
message.stream_id === stream_id &&
                message.topic.toLowerCase() === topic.toLowerCase()
```